nanopub/java_wrapper.py

```diff
@@ -33,7 +43,20 @@ def sign(self, unsigned_file: Union[str, Path]) -> str:
         return self._get_signed_file(unsigned_file)
 
     def publish(self, signed: str):
-        self._run_command(f'{NANOPUB_SCRIPT} publish ' + signed)
+        """ Publish.
+
+        Publish the signed nanopub to the nanopub server. Publishing to the real server depends
+        on nanopub-java, for the test server we do a simple POST request.
+
+        TODO: Use nanopub-java for publishing to test once it supports it.
+        """
+        if self.use_test_server:
+            headers = {'content-type': 'application/x-www-form-urlencoded'}
+            with open(signed, 'rb') as data:
+                r = requests.post(NANOPUB_TEST_SERVER, headers=headers, data=data)
+            r.raise_for_status()
```

Comment:
I don't really like this, but we don't have any other options for now.

Reply:
Potentially we could consider taking publishing out of the java wrapper completely, and having the ```nanopub``` python lib do it itself using ```requests```. We would then keep ```nanopub-java``` for things like making nanopubs 'trusty' etc. Could that help to santize this a little?

Reply:
i.e. publishing wouldn't require ```if self.use_test_server``` anymore, since you would always specify which URL to publish to yourself, rather than using the java tool

Reply:
Yeah, could be a good option to always do the post ourselves. But doesn't nanopub-java do some magic to make sure something is published to all servers?

Preferably as a user you don't want to specify where to publish to. You shouldn't want to care about that.

I would propose to leave it like this (and leave the TODO in there to use nanopub-java again). Or would you prefer doing the POST from python? @raar1

Reply:
I'm happy to leave it as a TODO as you suggest. I also would like to avoid putting that functionality into python unless we know exactly how the java tool is doing it. It's not a big problem, so we can worry about this another time.